awswrangler/catalog/_create.py

```diff
@@ -35,17 +35,10 @@ def _create_table(  # pylint: disable=too-many-branches,too-many-statements,too-
     table_input: Dict[str, Any],
     table_type: Optional[str],
     table_exist: bool,
-    projection_enabled: bool,
     partitions_types: Optional[Dict[str, str]],
     columns_comments: Optional[Dict[str, str]],
     transaction_id: Optional[str],
-    projection_types: Optional[Dict[str, str]],
-    projection_ranges: Optional[Dict[str, str]],
-    projection_values: Optional[Dict[str, str]],
-    projection_intervals: Optional[Dict[str, str]],
-    projection_digits: Optional[Dict[str, str]],
-    projection_formats: Optional[Dict[str, str]],
-    projection_storage_location_template: Optional[str],
+    projection_params: Optional[Dict[str, Any]],
```

Comment:
A bit of a crazy idea but I'm curious what you think. I wonder if we should make this a ProjectionParams data class or named tuple, and then we could have all of its members documented. So when someone used:
```python
projection_params=ProjectionParams(
  ranges=...,
  values=...,
)
```

This way, the parameters for projection would all be defined and explained in a single place. Plus users would get help from IntelliSense or whichever IDE they are using when filling out these parameters.

The major downside though is that it would mean this new class needs to be imported and instantiated. We could mitigate this issue, however, by making the type of `projection_params` the following:
```python
projection_params: Optional[Union[ProjectParams, Dict[str, Any]]]
```

Then, customers could make their code brief by using just the Python dict, which we would then silently convert to the ProjectionParams object.

Reply:
I do like the idea overall although this will put the burden of keeping `ProjectionParams` up-to-date on us. Also if we're changing it here we should do it for other parameters as well, e.g. s3, pyarrow etc... Keeping that whole pile up-to-date might be a challenge. Just purely passing dict seems a lot easier. Not sure what others think though

Reply:
I didn't realize we were merely forwarding these parameters into `boto`, I thought we were doing something else with them, which would require us to define them. You're right, it doesn't make sense for us to duplicate this definition.